packages/block-editor/src/store/selectors.js

```diff
@@ -2358,6 +2358,46 @@ const getAllAllowedPatterns = createSelector(
 	]
 );
 
+/**
+ * Returns whether there is at least one allowed pattern for inner blocks children.
+ * This is useful for deferring the parsing of all patterns until needed.
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional target root client ID.
+ *
+ * @return {boolean} If there is at least one allowed pattern.
+ */
+export const hasAllowedPatterns = createSelector(
```

Comment:
Let's make the selector experimental or private, just like all other pattern selectors are. The reason is block lazy loading (#51778): parsing is an async operation there, and a sync selector won't be able to parse anything. The entire pattern parsing approach will have to be reworked, hence `__experimental`.

Reply:
Sounds good.